ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -307,6 +346,63 @@ atPoint IdeOptions{} (HAR _ hf _ _ (kind :: HieKind hietype)) (DKMap dm km) env
             UnhelpfulLoc {} | isInternalName name || isSystemName name -> Nothing
             _ -> Just $ "*Defined " <> printOutputable (pprNameDefnLoc name) <> "*"
 
+        -- We want to render the root constraint even if it is a let,
+        -- but we don't want to render any subsequent lets
+        renderEvidenceTree :: Tree (EvidenceInfo a) -> SDoc
+        -- However, if the root constraint is simply a<n indirection (via let) to a single other constraint,
+        -- we can still skip rendering it
+        -- A 'let' indirection is conceptually every evidence that is acquired in a 'Let' statement.
+        -- As a very simple example, take:
+        --
+        -- @
+        --  foo = show 2.0
+        -- @
+        --
+        -- The evidence at 'show' is acquired in a 'Let' (i.e., the definition of 'foo').
```

Comment:
This is not correct. The evidence ghc generates is made up of a few primitives, like `WpLet` (let bindings, `WpEvLam` (lambda abstractions) and so on. The let binding refers to these lets.

For example, evidence for `Show ([Int], Bool)` might look like:

```haskell
$dShow,[]IntBool = $fShow,[]IntBool -- indirection, we don't gain anything by printing this
$fShow,[]IntBool = $dShow, $fShow[]Int $fShowBool -- This is the root "let" we render as a tree
$fShow[]Int = $dShow[] $fShowInt -- second level let, collapse it into its parent $fShow,[]IntBool
$fShowInt = base:Data.Int.$dShowInt -- indirection, remove it
$fShowBool = base:Data.Bool.$dShowBool -- indirection, remove it

in $dShow,[]IntBool
```

On doing this we end up with the tree `Show ([Int], Bool) -> (Show (,), Show [], Show Int, Show Bool)`

Reply:
Thanks, I will work this feedback in